discovery/event_test.go

```diff
@@ -0,0 +1,21 @@
+package discovery
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestEvent(t *testing.T) {
+	e := newEvent()
+	require.False(t, e.IsDone())
+	require.False(t, e.IsDone())
+
+	e.SetDone()
+	require.True(t, e.IsDone())
+	require.True(t, e.IsDone())
+
+	e.SetDone()
+	require.True(t, e.IsDone())
+	require.True(t, e.IsDone())
+}
```

Comment:
How come we're asserting on `IsDone` twice in each block?

Reply:
I trimmed this down. It was to make sure they are safe to call more than once.